src/mad_spark_alt/evolution/cost_estimator.py

```
@@ -0,0 +1,654 @@
+
+        # Calculate required reduction
+        reduction_needed = (current_cost - target_budget) / current_cost
+
+        # Suggest cheaper model
+        cheaper_models = [
+            ("gpt-3.5-turbo", GPT35_TURBO_COST_REDUCTION),
+            ("gemini-pro", GEMINI_PRO_COST_REDUCTION),
+            ("claude-3-sonnet", CLAUDE3_SONNET_COST_REDUCTION),
+        ]
+
+        for alt_model, reduction in cheaper_models:
```

Ruff (0.12.2)

Loop control variable `reduction` not used within loop body

Rename unused `reduction` to `_reduction`

(B007)